### PR TITLE
[Feature] Add shouldPerformAction method with a message parameter

### PR DESCRIPTION
**Pull request by KaterinaWire**
## What's new in this PR?

Implement a new `shouldPerformActionWithMessage` method that can use a string parameter to generate an warning message.

Needs releases with:

- [ wireapp/wire-ios-sync-engine/pull/1454 ]

### CODE REVIEW

File: Wire-iOS/Sources/URLActionRouter.swift [hunk 1]
```diff
@@ -148,8 +148,16 @@ extension URLActionRouter: PresentationDelegate {
...
+            decisionHandler(true)
+        }
+    }
+
+    func shouldPerformActionWithMessage(_ message: String, action: URLAction, decisionHandler: @escaping (_ shouldPerformAction: Bool) -> Void) {
+        typealias UrlAction = L10n.Localizable.UrlAction
+        switch action {
         case .joinConversation:
-            presentConfirmationAlert(title: UrlAction.title, message: UrlAction.JoinConversation.Confirmation.message(""), decisionHandler: decisionHandler)
+            presentConfirmationAlert(title: UrlAction.title, message: UrlAction.JoinConversation.Confirmation.message(message), decisionHandler: decisionHandler)
```
Comment (johnxnguyen) [inline Wire-iOS/Sources/URLActionRouter.swift:160]: We decided not to have a title for this confirmation, so it should be `nil`.

**Review by johnxnguyen — CHANGES_REQUESTED** [review]

File: Wire-iOS/Sources/URLActionRouter.swift (hunk 1 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/URLActionRouter.swift]: Fixed

**Review by johnxnguyen — APPROVED** [review]

File: Wire-iOS/Sources/URLActionRouter.swift [hunk 2]
```diff
@@ -148,8 +148,16 @@ extension URLActionRouter: PresentationDelegate {
         case .accessBackend(configurationURL: let configurationURL):
             guard SecurityFlags.customBackend.isEnabled else { return }
             presentCustomBackendAlert(with: configurationURL)
+        default:
+            decisionHandler(true)
+        }
+    }
+
+    func shouldPerformActionWithMessage(_ message: String, action: URLAction, decisionHandler: @escaping (_ shouldPerformAction: Bool) -> Void) {
+        typealias UrlAction = L10n.Localizable.UrlAction
```
Comment (David-Henner) [inline Wire-iOS/Sources/URLActionRouter.swift:157]: typealias might not be needed if it's only used once

**Review by David-Henner — APPROVED** [review]

File: Wire-iOS/Sources/URLActionRouter.swift (hunk 2 above)
Comment (KaterinaWire) [reply Wire-iOS/Sources/URLActionRouter.swift]: Fixed

**Pull request merged**